extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -28,139 +32,78 @@ require([
+	if (w.wgContentLanguage !== 'en') {
 		if (videosModule) {
 			videosModule('#recirculation-rail');
 		}
-		discussions();
+		// oldDiscussions();
 		return;
 	}
 
-	recircModules.forEach(function (recircModule) {
-		var deferred = $.Deferred();
-
-		var configuredHelper = liftigniter(recircModule.options);
-		configuredHelper.loadData()
-			.done(function (data) {
-				deferred.resolve(data);
-			})
-			.fail(function (err) {
-				deferred.reject(err);
-			});
 
-		// We need to keep track of the liftigniter helpers so we can setup tracking for them
-		liftigniterHelpers[recircModule.id] = configuredHelper;
-
-		if (recircModule.viewModule) {
-			views[recircModule.viewModule] = views[recircModule.viewModule] || [];
-			views[recircModule.viewModule].push(deferred);
-		}
+	// prepare & render right rail recirculation module
+	liftigniter.prepare(railRecirculation).done(function (data) {
+		require(['ext.wikia.recirculation.views.premiumRail'], function (viewFactory) {
+			viewFactory().render(data);
+		});
 	});
 
-	$.each(views, function (viewModule, promises) {
-		var deferred = $.Deferred();
-
-		completed.push(deferred);
-
-		log('Initializing View: ' + viewModule, 'info', logGroup);
-		require([viewModule], function (viewFactory) {
-			$.when.apply($, promises)
-				.done(function () {
-					var view = viewFactory(),
-						args = Array.prototype.slice.call(arguments),
-						data = {
-							title: '',
-							items: []
-						};
-
-					log(args, 'info', logGroup);
-
-					args.forEach(function (result) {
-						if (!result) {
-							return;
-						}
-
-						if (result.title && data.title.length === 0) {
-							data.title = result.title;
-						}
 
-						data.items = data.items.concat(result.items);
-					});
-
-					view.render(data)
-						.then(view.setupTracking())
-						.then(deferred.resolve);
-				})
-				.fail(handleError(viewModule), function () {
-					deferred.reject(viewModule);
-				});
+	// prepare & render mixed content footer module
+	var mixedContentFooterData = [
+		liftigniter.prepare(mixedContentFooter.nsItems),
+		liftigniter.prepare(mixedContentFooter.wikiItems),
+		discussions.prepare()
+	];
+	$.when.apply($, mixedContentFooterData).done(function (nsItems, wikiItems, discussions) {
+		require(['ext.wikia.recirculation.views.mixedFooter'], function (viewFactory) {
+			viewFactory().render({
+				nsItems: nsItems,
+				wikiItems: wikiItems,
+				discussions: discussions
+			});
 		});
 	});
 
-	$.when.apply($, completed)
-		.done(function () {
-			log('Finished rendering recirculation', 'info', logGroup);
+	// fetch data for all recirculation modules
+	// TODO lazy load some data on scroll
+	liftigniter.fetch('ns');
+	liftigniter.fetch('wiki');
+	discussions.fetch();
 
-			$.each(liftigniterHelpers, function (viewModule, helper) {
-				helper.setupTracking();
-			});
-		})
-		.fail(function (viewModule) {
-			log('Error running recirc at: ' + viewModule, 'info', logGroup);
-		});
+	// TODO handle errors
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)